### PR TITLE
[TFLite] Support quantized GREATER op in TFLite frontend

### PR DESCRIPTION
**Pull request by dchauhan-arm**
Support GREATER quantization operation conversion as part of issue #9187

### CODE REVIEW

File: tests/python/frontend/tflite/test_forward.py [hunk 1]
```diff
@@ -2486,9 +2492,9 @@ def test_all_elemwise():
     _test_forward_elemwise_quantized(_test_maximum)
     _test_forward_elemwise(_test_minimum)
     _test_forward_elemwise_quantized(_test_minimum)
-    _test_forward_elemwise(_test_greater)
     _test_forward_elemwise(_test_squared_difference)
     _test_forward_elemwise(_test_greater_equal)
+    _test_forward_elemwise(_test_greater)
```
Comment (lhutton1) [inline tests/python/frontend/tflite/test_forward.py:2497]: Do we also need a call to `_test_forward_elemwise_quantized` to test the quantized version of "greater"?

Comment (dchauhan-arm) [reply tests/python/frontend/tflite/test_forward.py:2497]: That is correct, got a patch in the pipes to get that in, a couple functions need tweaking for comparison operators like these.

**Comment by dchauhan-arm** [comment]
cc @leandron and @lhutton1 for a second look 🙂

**Comment by lhutton1** [comment]
If I inspect the Relay graph(s) that is tested when running `_test_forward_elemwise_quantized(_test_greater, True)`, it seems as though it doesn't get quantized in any way e.g.
```
def @main(%inq_0: Tensor[(3, 6), float32] /* ty=Tensor[(3, 6), float32] */, %inq_1: Tensor[(3, 6), float32] /* ty=Tensor[(3, 6), float32] */, output_tensor_names=["Greater"]) -> Tensor[(3, 6), bool] {
  greater(%inq_0, %inq_1) /* ty=Tensor[(3, 6), bool] */
}
```
So then it seems we're still only testing the float32 implementation of the operator? For a quantized operator I'd expect int/uint inputs being passed to the `greater` op - maybe I missed something?

**Review by lhutton1 — CHANGES_REQUESTED** [review]
If I inspect the Relay graph(s) that is tested when running `_test_forward_elemwise_quantized(_test_greater, True)`, it seems as though it doesn't get quantized in any way e.g.
```
def @main(%inq_0: Tensor[(3, 6), float32] /* ty=Tensor[(3, 6), float32] */, %inq_1: Tensor[(3, 6), float32] /* ty=Tensor[(3, 6), float32] */, output_tensor_names=["Greater"]) -> Tensor[(3, 6), bool] {
  greater(%inq_0, %inq_1) /* ty=Tensor[(3, 6), bool] */
}
```
So then it seems we're still only testing the float32 implementation of the operator? For a quantized operator I'd expect int/uint inputs being passed to the `greater` op

**Review by leandron — COMMENTED** [review]
@dchauhan-arm just a friendly ping to update this PR with quantised testing for the operator.

**Comment by dchauhan-arm** [comment]
> @dchauhan-arm just a friendly ping to update this PR with quantised testing for the operator.

yes, what I think I'll do is rewrite a test specifically for the greater op, and compare output of tvm against tflite.

**Comment by leandron** [comment]
@lhutton1 can you have another look on this one?

**Comment by dchauhan-arm** [comment]
cc @lhutton1 for a second (second) look

File: tests/python/frontend/tflite/test_forward.py [hunk 2]
```diff
@@ -2480,26 +2493,15 @@ def _test_forward_elemwise(testop):
...
-        # no need for fake_quant to hold tensors in float32 until conversion
-        testop(
-            [
-                np.array(np.random.uniform(0, 255, (3, 6)), dtype=np.float32),
-                np.array(np.random.uniform(0, 255, (3, 6)), dtype=np.float32),
-            ],
-            quantized=True,
-            qnn_op=testop,
-        )
+def _test_forward_elemwise_quantized(testop, comparison_op=False):
```
Comment (lhutton1) [inline tests/python/frontend/tflite/test_forward.py:2496]: Nit: looks like `comparison_op` here is not used, can we remove it?

**Review by lhutton1 — APPROVED** [review]
Thanks for persisting with this @dchauhan-arm, the quantized graph we are testing looks better now! Just a small nit, but otherwise LGTM :)

File: tests/python/frontend/tflite/test_forward.py (hunk 2 above)
Comment (dchauhan-arm) [reply tests/python/frontend/tflite/test_forward.py:2496]: good spot, thanks

**Comment by lhutton1** [comment]
hmmm looks like there's a seg fault happening in `test_elemwise[_test_greater]` on aarch64 CI, so maybe there is an issue with this version of tensorflow on aarch64

**Comment by NicolaLancellotti** [comment]
With tflite 2.9.1 `test_elemwise[_test_greater]` does not fail any more. We should wait for #12130 and #12131 to be merged.

**Comment by leandron** [comment]
This is now unblocked by https://github.com/apache/tvm/pull/12738 - @dchauhan-arm can you remove the conflict and resubmit this for CI to check?

**Pull request closed without merging**